### ISSUE TITLE
Implement chat bans

### ISSUE THREAD

**Issue by 2Pacalypse-** [ShieldBattery/ShieldBattery#1066]
I've been increasingly getting messages from people complaining about bad manners directed at them. While we usually just banned people for the more egregious messages, some of this bad manners is just the typical shit talking where an outright ban might seem a bit harsh.

Having an option to mute these shit talkers for a certain amount of time might be a more preferable way to make the chat channels a better experience for everyone. Although having a home page would help with this as well, so people who are particularly sensitive to this kind of shit talking can just leave all of the chat channels and play in peace.

Few open questions regarding this feature:
- should the channel owners/moderators be able to mute people in their channels or is this only a server admin feature?
- should we have a per-channel mute or only server-wide mute?
- should we add a whisper-wide mute?
- should we have a time-based mute or rely on manual unmuting?
- should we add a reason message for muting?
- should this be integrated with the game as well so people can't chat in game either, or is this supposed to be a separate thing? muting people in-game might be a bit more complicated since they should still be able to message their allies?

**Comment by tec27** [comment]
> * should the channel owners/moderators be able to mute people in their channels or is this only a server admin feature?

No, this is account-based not channel-based. If people don't want a certain person talking in their channel, they can just ban them

> * should we have a per-channel mute or only server-wide mute?

Server-wide

> * should we add a whisper-wide mute?

imo this is a "chat ban", it should affect all forms of chat (including whisper and ingame)

> * should we have a time-based mute or rely on manual unmuting?

time-based

> * should we add a reason message for muting?

May as well, the table for this needs to be similar to the ban table anyway.

> * should this be integrated with the game as well so people can't chat in game either, or is this supposed to be a separate thing? muting people in-game might be a bit more complicated since they should still be able to message their allies?

Yes this should be integrated with the game, this is why I haven't built this feature yet. If people want to be able to talk to their allies they should try to behave so they don't get banned, so I don't have much sympathy for them. We should probably notify people in the game that they are chat-banned in some way though.

**Comment by tec27** [comment]
Implemented in 10.x
